pkgs/applications/science/math/sage/threejs-sage.nix

```diff
@@ -0,0 +1,18 @@
+{ stdenv, fetchFromGitHub }:
+
+stdenv.mkDerivation rec {
+  pname = "threejs-sage";
+  version = "r122";
+
+  src = fetchFromGitHub {
```

Comment:
It would be nice to build this from the upstream sources. This is probably more practical though, given the node packaging difficulties.

Reply:
I agree, we can and should build it from upstream sources! It was my original plan and I think I know how to do it (stealing a few tricks from the `rust-analyzer` VS Code extension package). I just didn't find the time to finish it and thought this would be a good enough stopgap given that two users complained about it (one on IRC, one in #118806) :) I'll have some more time next month to do this. But I think we would probably need to pin to a specific version of upstream threejs, since they're doing a ton of refactorings to use ES6 Modules and it would be too much of a moving target.